File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/ml/BarcodeScanner.kt
package dev.hebei.camerax_android.legacy.ml

import com.google.android.gms.tasks.Task
import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.PigeonApiBarcodeScanner

class BarcodeScanner(registrar: CameraXRegistrar) : PigeonApiBarcodeScanner(registrar) {
    override fun process0(
        pigeon_instance: Stub,
        image: com.google.android.odml.image.MlImage,
        callback: (Result<List<com.google.mlkit.vision.barcode.common.Barcode>>) -> Unit
    ) {
        pigeon_instance.process(image).addOnSuccessListener { value ->
            callback(Result.success(value))
        }.addOnFailureListener { exception ->
            callback(Result.failure(exception))
        }
    }

    override fun process1(
        pigeon_instance: Stub,
        image: com.google.mlkit.vision.common.InputImage,
        callback: (Result<List<com.google.mlkit.vision.barcode.common.Barcode>>) -> Unit
    ) {
        pigeon_instance.process(image).addOnSuccessListener { value ->
            callback(Result.success(value))
        }.addOnFailureListener { exception ->
            callback(Result.failure(exception))
        }
    }

    class Stub internal constructor(override val obj: com.google.mlkit.vision.barcode.BarcodeScanner) :
        Detector.Stub(obj) {
        fun process(image: com.google.android.odml.image.MlImage): Task<List<com.google.mlkit.vision.barcode.common.Barcode>> {
            return obj.process(image)
        }

        fun process(image: com.google.mlkit.vision.common.InputImage): Task<List<com.google.mlkit.vision.barcode.common.Barcode>> {
            return obj.process(image)
        }
    }
}

File: camerax/lib/src/ml/barcode/barcode_scanner.dart
import 'package:camerax/src/ml/detector.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;
import 'package:meta/meta.dart';

import 'barcode_scanner_options.dart';

final class BarcodeScanner extends Detector<List<$interface.Barcode>> {
  final $interface.BarcodeScanner _obj;

  BarcodeScanner({
    BarcodeScannerOptions? options,
  }) : _obj = $interface.BarcodeScanner(
          options: options?.obj,
        );

  @internal
  @override
  $interface.BarcodeScanner get obj => _obj;
}
